src/geometry/euclidean/ceva_theorem.lean

```diff
@@ -0,0 +1,325 @@
+/-
+Copyright (c) 2021 Mantas Bakšys. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mantas Bakšys
+-/
+import tactic.linarith
+import tactic.fin_cases
+import tactic.norm_num
+import geometry.euclidean
+import linear_algebra.affine_space.basis
+import tactic.basic
+import analysis.normed_space.add_torsor_bases
+import data.equiv.basic
+import group_theory.perm.list
+
+/-!
+# Ceva's Theorem
+
+This file (although could be merged to another file later) proves Ceva's Theorem in Euclidean
+Geometry following the barycentric coordinate proof. For a reference,
+see https://en.wikipedia.org/wiki/Ceva%27s_theorem.
+
+Because of the currently generality that `interior_convex_hull_aff_basis` is stated in, we prove
+Ceva's theorem in it's current form. This will need to be upgraded to the usual setting of
+Euclidean Geometry results after the affinity refactor.
+
+**This is also a Problem 61 on Freek's list**
+-/
+
+variables {V : Type*} [inner_product_space ℝ V] [normed_space ℝ V]
+include V
+
+open affine finite_dimensional finset equiv affine_basis fintype basis
+
+open_locale euclidean_geometry big_operators
+
+lemma affine_basis.vsub_eq_coord_smul_sum (O : V) {ι : Type*} [fintype ι]
+  (S : affine_basis ι ℝ V) : ∀ X : V, X -ᵥ O = ∑ (i : ι), S.coord i O • (X -ᵥ S.points i) :=
+begin
+  set w : ι → ℝ := λ i, S.coord i O with hw,
+  have hs := sum_coord_apply_eq_one S O,
+  intro X,
+  have h₄ := weighted_vsub_of_point_vadd_eq_of_sum_eq_one univ w S.points hs X O,
+  rw vadd_eq_vadd_iff_sub_eq_vsub at h₄,
+  rw ← h₄,
+  have h₅ : (univ.weighted_vsub_of_point S.points O) w = (0 : V),
+  { suffices h : univ.affine_combination S.points w = O,
+    { rw affine_combination_eq_weighted_vsub_of_point_vadd_of_sum_eq_one _ w S.points hs O at h,
+      symmetry' at h,
+      rw eq_vadd_iff_vsub_eq at h,
+      rw ← h,
+      exact vsub_self O},
+    convert affine_combination_coord_eq_self S O },
+  rw h₅,
+  simp only [vsub_eq_sub, zero_sub, weighted_vsub_of_point_apply, neg_eq_iff_add_eq_zero,
+      ← sum_add_distrib, smul_sub, smul_sub],
+  norm_num,
+end
+
+lemma equiv.sum_coords_congr {ι ι' : Type*} [fintype ι] [fintype ι'] (e : ι ≃ ι') (b : basis ι ℝ V)
+  (c : basis ι' ℝ V) (h : c = b.reindex e) : b.sum_coords = c.sum_coords :=
+begin
+  ext x,
+  simp only [coe_sum_coords_of_fintype, h, reindex_repr, reindex, basis.coord_apply,
+  linear_equiv.trans_apply, finsupp.dom_congr_apply, fintype.sum_apply,
+  finsupp.equiv_map_domain_apply, finsupp.dom_lcongr_apply],
+  rw sum_equiv e,
+  intro i,
+  rw symm_apply_apply
+end
+
+lemma affine_basis.coord_perm {ι : Type*} [fintype ι] (σ : perm ι) {i : ι}
+  (S T : affine_basis ι ℝ V) (O : V) (h : T.points = S.points ∘ σ) :
+  S.coord (σ i) O = T.coord i O :=
```

Comment:
I think what we actually need here in mathlib is an API for `affine_basis.reindex`, to go in `linear_algebra.affine_space.basis`, using the API for `basis.reindex` as a guide (including working for arbitrary equivs between two possibly different index types). Then you'd only have one basis as an argument to the version of this lemma that goes in that API, and it would say `(b.reindex e).coord = b.coord ∘ e.symm` like the one for linear bases.